cognite/neat/session/_session/_data_model/_template.py

```diff
@@ -0,0 +1,205 @@
+from pathlib import Path
+from typing import Any, Literal
+
+from cognite.client.data_classes.data_modeling import DataModelIdentifier
+
+from cognite.neat.core._constants import BASE_MODEL
+from cognite.neat.core._data_model._shared import ImportedDataModel
+from cognite.neat.core._data_model.exporters import ExcelExporter
+from cognite.neat.core._data_model.importers import ExcelImporter
+from cognite.neat.core._data_model.models import UnverifiedConceptualDataModel
+from cognite.neat.core._data_model.models._base_verified import RoleTypes
+from cognite.neat.core._data_model.models.physical import PhysicalValidation
+from cognite.neat.core._data_model.transformers import (
+    AddCogniteProperties,
+    IncludeReferenced,
+    ToDataProductModel,
+    ToEnterpriseModel,
+    VerifiedDataModelTransformer,
+)
+from cognite.neat.core._issues import IssueList, catch_issues
+from cognite.neat.core._utils.reader import NeatReader, PathReader
+from cognite.neat.session._state import SessionState
+from cognite.neat.session.exceptions import NeatSessionError, session_class_wrapper
+
+
+@session_class_wrapper
+class TemplateAPI:
```

Comment:
As you are refactoring this, maybe it can be the opportunity to split the responsibility here. The TemplateAPI is handling "stateful" and "stateless" scenarios, so it's a good candidate to split. 

`enterprise_model` and `data_product_model` are reusing the current session data model, from the current state

`conceptual_model` and `expand` are reading from file, so they are kinda self contained. Maybe on domain level makes sense to keep both together... But seperation of concerns might help to extend this later on.